src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -273,12 +273,16 @@ public void init(AdaptorContext context) throws Exception {
 
     // Verify all column names.
     try (Connection conn = makeNewConnection()) {
-      verifyColumnNames(conn, "db.everyDocIdSql", everyDocIdSql,
-          "db.uniqueKey", uniqueKey.getDocIdSqlColumns());
-      verifyColumnNames(conn, "db.singleDocContentSql", singleDocContentSql,
-          "db.singleDocContentSqlParameters", uniqueKey.getContentSqlColumns());
-      verifyColumnNames(conn, "db.aclSql", aclSql,
-          "db.aclSqlParameters", uniqueKey.getAclSqlColumns());
+      uniqueKey.addColumnTypes(
+          verifyColumnNames(conn, "db.everyDocIdSql", everyDocIdSql,
+              "db.uniqueKey", uniqueKey.getDocIdSqlColumns()));
+      uniqueKey.addColumnTypes(
```

Comment:
I think these next two are redundant. All of the columns have to appear in db.uniqueKey, and if the type map is case-sensitive, then all of the case-sensitive keys have to appear there as well.